components/PlantForm.js

```diff
@@ -63,28 +67,44 @@ const fertiliserOptions = [
   { id: "fertiliserSeason4", value: "Winter", label: "Winter" },
 ];
 
-export default function PlantForm({ onCreatePlant }) {
+export default function PlantForm({
+    onCreatePlant,
+    onEditPlant,
+    isEditMode = false,
+    initialData = {}
+ }) {
+
   function handleSubmit(event) {
     event.preventDefault();
 
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
 
     const selectedSeasons = formData.getAll("fertiliserSeason");
+    data.fertiliserSeason = selectedSeasons;
 
     if (selectedSeasons.length === 0) {
       // FYI: Hinzufügen einer universellen Error-Message für alle Pflichtfelder folgt
       alert("Please select at least one season.");
       return;
     }
-    data.fertiliserSeason = selectedSeasons;
-    onCreatePlant(data);
+
+    if(isEditMode) {
+      onEditPlant({ ...initialData, ...data });
+    } else {
+      onCreatePlant({ ...initialData, ...data });
+    }
 
     event.target.reset();
   }
 
+  function handleCancel() {
+    window.location.reload();
+  }
```

Comment:
Instead of using `window.location.reload() ` consider  to control form visibility without reloading, which is smoother for the user. See suggestions above.

```suggestion
```